File: odemapster-r2rml-rdb-querytranslator/src/main/java/es/upm/fi/dia/oeg/obdi/wrapper/r2rml/rdb/querytranslator/engine/RunnerFactory.java
package es.upm.fi.dia.oeg.obdi.wrapper.r2rml.rdb.querytranslator.engine;

import es.upm.fi.dia.oeg.obdi.core.ConfigurationProperties;
import es.upm.fi.dia.oeg.obdi.core.engine.IQueryTranslationOptimizer;
import es.upm.fi.dia.oeg.obdi.core.engine.IQueryTranslator;
import es.upm.fi.dia.oeg.obdi.core.querytranslator.QueryTranslationOptimizerFactory;
import es.upm.fi.dia.oeg.obdi.wrapper.r2rml.rdb.engine.R2RMLRunner;

public class RunnerFactory {
	public static R2RMLRunner createR2RMLRunnerC(String configurationDirectory, String configurationFile)
			throws Exception {
		ConfigurationProperties configurationProperties = 
				new ConfigurationProperties(configurationDirectory, configurationFile);
		R2RMLRunner r2rmlRunner = RunnerFactory.createR2RMLRunnerC(configurationProperties);
		return r2rmlRunner;
	}

	public static R2RMLRunner createR2RMLRunnerC(ConfigurationProperties configurationProperties)
			throws Exception {
		R2RMLRunner r2rmlRunner = new R2RMLRunner(configurationProperties);
		r2rmlRunner.buildQueryTranslator();
		IQueryTranslator queryTranslator = r2rmlRunner.getQueryTranslator();
		IQueryTranslationOptimizer queryTranslationOptimizerC = 
				QueryTranslationOptimizerFactory.createQueryTranslationOptimizerC();
		queryTranslator.setOptimizer(queryTranslationOptimizerC);
		return r2rmlRunner;
	}

	public static R2RMLRunner createR2RMLRunnerE(String configurationDirectory, String configurationFile)
			throws Exception {
		ConfigurationProperties configurationProperties = 
				new ConfigurationProperties(configurationDirectory, configurationFile);
		R2RMLRunner r2rmlRunner = RunnerFactory.createR2RMLRunnerE(configurationProperties);
		return r2rmlRunner;
	}
	
	public static R2RMLRunner createR2RMLRunnerE(ConfigurationProperties configurationProperties)
			throws Exception {
		R2RMLRunner r2rmlRunner = new R2RMLRunner(configurationProperties);
		r2rmlRunner.buildQueryTranslator();
		IQueryTranslator queryTranslator = r2rmlRunner.getQueryTranslator();
		IQueryTranslationOptimizer queryTranslationOptimizerE = 
				QueryTranslationOptimizerFactory.createQueryTranslationOptimizerE();
		queryTranslator.setOptimizer(queryTranslationOptimizerE);
		return r2rmlRunner;
	}

	public static R2RMLRunner createR2RMLRunnerFC(String configurationDirectory, String configurationFile)
			throws Exception {
		ConfigurationProperties configurationProperties = 
				new ConfigurationProperties(configurationDirectory, configurationFile);
		R2RMLRunner r2rmlRunner = RunnerFactory.createR2RMLRunnerFC(configurationProperties);
		return r2rmlRunner;
	}

	public static R2RMLRunner createR2RMLRunnerFC(ConfigurationProperties configurationProperties)
			throws Exception {
		R2RMLRunner r2rmlRunner = new R2RMLRunner(configurationProperties);
		r2rmlRunner.buildQueryTranslator();
		IQueryTranslator queryTranslator = r2rmlRunner.getQueryTranslator();
		IQueryTranslationOptimizer queryTranslationOptimizerFC = 
				QueryTranslationOptimizerFactory.createQueryTranslationOptimizerFC();
		queryTranslator.setOptimizer(queryTranslationOptimizerFC);
		return r2rmlRunner;
	}
	
	public static R2RMLRunner createR2RMLRunnerFE(String configurationDirectory, String configurationFile)
			throws Exception {
		ConfigurationProperties configurationProperties = 
				new ConfigurationProperties(configurationDirectory, configurationFile);
		R2RMLRunner r2rmlRunner = RunnerFactory.createR2RMLRunnerFE(configurationProperties);
		return r2rmlRunner;
	}

	public static R2RMLRunner createR2RMLRunnerFE(ConfigurationProperties properties)
			throws Exception {
		R2RMLRunner r2rmlRunner = new R2RMLRunner(properties);
		r2rmlRunner.buildQueryTranslator();
		IQueryTranslator queryTranslator = r2rmlRunner.getQueryTranslator();
		IQueryTranslationOptimizer queryTranslationOptimizerFE = 
				QueryTranslationOptimizerFactory.createQueryTranslationOptimizerFE();
		queryTranslator.setOptimizer(queryTranslationOptimizerFE);
		return r2rmlRunner;
	}

}

File: morph-r2rml-rdb/src/main/scala/es/upm/fi/dia/oeg/morph/r2rml/rdb/engine/RunnerFactory.scala
package es.upm.fi.dia.oeg.morph.r2rml.rdb.engine

import es.upm.fi.dia.oeg.morph.base.MorphProperties
import es.upm.fi.dia.oeg.morph.base.engine.QueryTranslationOptimizerFactory
import es.upm.fi.dia.oeg.morph.base.engine.MorphBaseRunner

class RunnerFactory {

}

object RunnerFactory {
	def createR2RMLRunnerC(configurationDirectory:String , configurationFile:String) 
	: MorphBaseRunner = {
		val properties = MorphProperties.apply(configurationDirectory, configurationFile);
		val r2rmlRunner = RunnerFactory.createR2RMLRunnerC(properties);
		return r2rmlRunner;
	}

	def createR2RMLRunnerC(properties:MorphProperties) 
	: MorphBaseRunner = {
		val runnerFactory = new R2RMLRDBRunnerFactory();
		val r2rmlRunner = runnerFactory.createRunner(properties);
		val queryTranslator = r2rmlRunner.queryTranslator
		if(queryTranslator.isDefined) {
			val queryTranslationOptimizerC = 
					QueryTranslationOptimizerFactory.createQueryTranslationOptimizerC();
			queryTranslator.get.optimizer = queryTranslationOptimizerC;		  
		}

		r2rmlRunner;
	}

	def createR2RMLRunnerE(configurationDirectory:String, configurationFile:String) : MorphBaseRunner = {
		val properties = 
				MorphProperties.apply(configurationDirectory, configurationFile);
		val r2rmlRunner = RunnerFactory.createR2RMLRunnerE(properties);
		r2rmlRunner;
	}
	
	def  createR2RMLRunnerE(properties:MorphProperties) : MorphBaseRunner = {
		val runnerFactory = new R2RMLRDBRunnerFactory();
		val r2rmlRunner = runnerFactory.createRunner(properties);
//		r2rmlRunner.buildQueryTranslator();
		val queryTranslator = r2rmlRunner.queryTranslator;
		if(queryTranslator.isDefined) {
			val queryTranslationOptimizerE = 
					QueryTranslationOptimizerFactory.createQueryTranslationOptimizerE();
			queryTranslator.get.optimizer = queryTranslationOptimizerE;  
		}

		return r2rmlRunner;
	}

	def createR2RMLRunnerFC(configurationDirectory:String, configurationFile:String) : MorphBaseRunner = {
		val properties = 
				MorphProperties.apply(configurationDirectory, configurationFile);
		val r2rmlRunner = RunnerFactory.createR2RMLRunnerFC(properties);
		r2rmlRunner;
	}

	def createR2RMLRunnerFC(properties:MorphProperties ) : MorphBaseRunner = {
		val runnerFactory = new R2RMLRDBRunnerFactory();
		val r2rmlRunner = runnerFactory.createRunner(properties);
//		r2rmlRunner.buildQueryTranslator();
		val queryTranslator = r2rmlRunner.queryTranslator;
		if(queryTranslator.isDefined) {
			val queryTranslationOptimizerFC = 
					QueryTranslationOptimizerFactory.createQueryTranslationOptimizerFC();
			queryTranslator.get.optimizer = queryTranslationOptimizerFC;		  
		}

		r2rmlRunner;
	}
	
	def createR2RMLRunnerFE(configurationDirectory:String, configurationFile:String) : MorphBaseRunner = {
		val properties = 
				MorphProperties.apply(configurationDirectory, configurationFile);
		val r2rmlRunner = RunnerFactory.createR2RMLRunnerFE(properties);
		r2rmlRunner;
	}

	def createR2RMLRunnerFE(properties:MorphProperties ) : MorphBaseRunner = {
		val runnerFactory = new R2RMLRDBRunnerFactory();
		val r2rmlRunner = runnerFactory.createRunner(properties);
//		r2rmlRunner.buildQueryTranslator();
		val queryTranslator = r2rmlRunner.queryTranslator;
		if(queryTranslator.isDefined) {
			val queryTranslationOptimizerFE = 
					QueryTranslationOptimizerFactory.createQueryTranslationOptimizerFE();
			queryTranslator.get.optimizer = queryTranslationOptimizerFE;		  
		}

		r2rmlRunner;
	}
  
}